Resources/Prototypes/Entities/Structures/Doors/Windoors/clockwork.yml

```diff
@@ -0,0 +1,141 @@
+      map: ["enum.DoorVisualLayers.BaseUnlit"]
+    - state: welded
+      map: ["enum.WeldableLayers.BaseWelded"]
+    - state: bolted_unlit
+      shader: unshaded
+      map: ["enum.DoorVisualLayers.BaseBolted"]
+    - state: emergency_unlit
+      shader: unshaded
+      map: ["enum.DoorVisualLayers.BaseEmergencyAccess"]
+    - state: panel_open
+      map: ["enum.WiresVisualLayers.MaintenancePanel"]
+  - type: AnimationPlayer
+  - type: ApcPowerReceiver
+  - type: ExtensionCableReceiver
+  - type: DoorSignalControl
+  - type: DeviceNetwork
+    deviceNetId: Wireless
+    receiveFrequencyId: BasicDevice
+  - type: WirelessNetworkConnection
+    range: 200
+  - type: DeviceLinkSink
+    ports:
+      - Open
+      - Close
+      - Toggle
+      - AutoClose
+      - DoorBolt
+  - type: DeviceLinkSource
+    ports:
+    - DoorStatus
+  - type: Damageable
+    damageContainer: Inorganic
+    damageModifierSet: Glass
+  - type: ExaminableDamage
+    messages: WindowMessages
+  - type: Destructible
+    thresholds:
+    - trigger:
+        !type:DamageTrigger
+        damage: 400
+      behaviors: #excess damage (nuke?). avoid computational cost of spawning entities.
+      - !type:DoActsBehavior
+        acts: [ "Destruction" ]
+    - trigger:
+        !type:DamageTrigger
+        damage: 200
+      behaviors:
+      - !type:SpawnEntitiesBehavior
+        spawn:
+          ShardGlassClockwork:
+            min: 1
+            max: 2
+          SheetBrass1:
+            min: 2
+            max: 4
+      - !type:DoActsBehavior
+        acts: [ "Destruction" ]
+  - type: AccessReader
+  - type: ContainerFill
+    containers:
+      board: [ DoorElectronics ]
+  - type: ContainerContainer
+    containers:
+      board: !type:Container
+  - type: Door
+    canCrush: false
+    openingAnimationTime: 0.9
+    closingAnimationTime: 0.9
+    openSound:
+      path: /Audio/Machines/windoor_open.ogg
+    closeSound:
+      path: /Audio/Machines/windoor_open.ogg
+    denySound:
+      path: /Audio/Machines/airlock_deny.ogg
+  - type: Airlock
+    keepOpenIfClicked: true
+    openPanelVisible: true
+    denyAnimationTime: 0.4
+    animatePanel: false
+    openUnlitVisible: true
+    # needed so that windoors will close regardless of whether there are people in it; it doesn't crush after all
+    safety: false
+  - type: NavMapDoor
+  - type: DoorBolt
+  - type: Electrified
+    enabled: false
+    usesApcPower: true
+  - type: WiresPanel
+  - type: Wires
+    boardName: wires-board-name-windoor
+    layoutId: Airlock
+  - type: UserInterface
+    interfaces:
+    - key: enum.WiresUiKey.Key
+      type: WiresBoundUserInterface
+  - type: Appearance
+  - type: WiresVisuals
+  - type: Airtight
+    fixVacuum: true
+    noAirWhenFullyAirBlocked: false
+    airBlockedDirection:
+      - South
+  - type: Construction
+    graph: Windoor
+    node: windoorClockwork
+    containers:
+    - board
+  - type: StaticPrice
+    price: 150
+  - type: PryUnpowered
```

Comment:
You really don't need 100 components here for 1 clockwork door mate, please make it re-useable as this will just be unmaintainable in a year.

Reply:
done, thanks for pointing this out cause i'd never have figured it out by myself 👍